File: ticket_ms/ticket_ms/report/nmpi/nmpi.js
// Copyright (c) 2023, Simon Wanyama and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["NMPI"] = {
	"filters": [
		{
			"fieldname": "name",
			"label": __("NMPI"),
			"fieldtype": "Data",
			"width": 100,
			"reqd": 0,
		},
		{
			"fieldname": "site_name",
			"label": __("Site"),
			"fieldtype": "Link",
			"options": "Site",
			"width": 100,
			"reqd": 0,
		},
		{
			"fieldname": "raise_date",
			"label": __("Date"),
			"fieldtype": "Date",
			"width": 100,
			"reqd": 0,
		},
		{
			"fieldname": "nmpi_action",
			"label": __("NMPI Action"),
			"fieldtype": "Select",
			"options": ["", "Rectified At Site", "Training Given At Site", "Corrective Action Required"],
			"width": 100,
			"reqd": 0,
		},
		{
			"fieldname": "nmpi_category",
			"label": __("NMPI Category"),
			"fieldtype": "Link",
			"options": "NMPI Category",
			"width": 100,
			"reqd": 0,
		},
		{
			"fieldname": "state",
			"label": __("Region/State"),
			"fieldtype": "Link",
			"options": "India State",
			"width": 100,
			"reqd": 0,
		},
		{
			"fieldname": "city",
			"label": __("City"),
			"fieldtype": "Data",
			"width": 100,
			"reqd": 0,
		},
	]
};

File: ticket_ms/ticket_ms/report/nmpi/nmpi.py
# Copyright (c) 2023, Simon Wanyama and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
	return get_columns(), get_data(filters)

def get_data(filters):

	# conditions
	conditions = []
	
	if filters.get("name"): conditions.append(["name", "=", filters.get("name")])
	if filters.get("site_name"): conditions.append(["site_name", "=", filters.get("site_name")])
	if filters.get("raise_date"): conditions.append(["raise_date", "=", filters.get("raise_date")])
	if filters.get("nmpi_action"): conditions.append(["nmpi_action", "=", filters.get("nmpi_action")])
	if filters.get("nmpi_category"): conditions.append(["nmpi_category", "=", filters.get("nmpi_category")])
	if filters.get("state"): conditions.append(["state", "=", filters.get("state")])
	if filters.get("city"): conditions.append(["city", "=", filters.get("city")])


	# Data
	data = frappe.db.get_all('NMPI Ticket', [
		"name",
		"raise_date",
		"site_name",
		"type",
		"state",
		"city",
		"nmpi_category",
		"nmpi_action",
		"description",
		"comment",
		"upload_photos_videos_or_documents"
	], filters=conditions)
	return data

def get_columns():
	return [
		"Name:Data:170",
		"Raise Date:Date:150",
		"Site Name:Link/Site:170",
		"Type:Select:170",
		"State:Link/India State:120",
		"City:Data:120",
		"NMPI Category:Link/NMPI Category:170",
		"NMPI Action:Select:130",
		"Description:Text:150",
		"Comment:Small Text:150"
		"Company:Link/Company:100"
		"Attachments:Attach:200"
	]

